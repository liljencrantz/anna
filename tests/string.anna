
def main(){

    /* Test replacing substring with other string of same length */
    print("Test 1\n");
    a := "aaabbbccc".thaw;
    a[3..6] = "ddd";
    print(a,"\n");

    /* Test replacing substring with other string of different length */
    print("Test 2\n");
    a = "aaabbbccc".thaw;
    a[3..6] = "dddEEE";
    print(a,"\n");

    /* Test replacing substring past end of original string */
    print("Test 3\n");
    a = "aaabbbccc".thaw;
    a[7..15] = "eeeee";
    print(a,"\n");

    /* Test replacing substring completely past original string */
    print("Test 4\n");
    a = "a".thaw;
    a[7..15] = "eeeee";
    a.each(idx, val){print(idx, ": ", val.ordinal, "\n")}

    /* Test replacing non-consecutive characters */
    print("Test 5\n");
    a = "aaabbbccc".thaw;
    a[1..7|2] = "qqq";
    print(a,"\n");

    /* Try replacing non-consecutive characters with wrong length of replacement. This should be a no-op. */
    print("Test 6\n");
    a = "aaabbbccc".thaw;
    a[1..7|2] = "qqqaaa";
    print(a,"\n");

    /* Test replacing non-consecutive characters past end of original string. */
    print("Test 7\n");
    a = "aaabbbccc".thaw;
    a[1..15|10] = "qq";
    a.each(idx, val){print(idx, ": ", val.ordinal, "\n")}

    /* Test replacing by index from end */
    print("Test 8\n");
    a="abc".thaw;
    a[-1] = 'C';
    print(a);
    print("\n");    

    /* Test replacing ranges by index from end */
    print("Test 9\n");
    a="abc".thaw;
    a[-2..-1] = "BC";
    print(a);
    print("\n");
}
