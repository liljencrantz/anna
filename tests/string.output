Test 1
aaadddccc
Test 2
aaadddEEEccc
Test 3
aaabbbceeeee
Test 4
0: 97
1: 0
2: 0
3: 0
4: 0
5: 0
6: 0
7: 101
8: 101
9: 101
10: 101
11: 101
Test 5
aqaqbqccc
Test 6
aaabbbccc
Test 7
0: 97
1: 113
2: 97
3: 98
4: 98
5: 98
6: 99
7: 99
8: 99
9: 0
10: 0
11: 113
Test 8
abC
Test 9
aBCc
