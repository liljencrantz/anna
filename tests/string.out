aaadddccc
aaadddEEEccc
aaabbbceeeee
0: 97
1: 0
2: 0
3: 0
4: 0
5: 0
6: 0
7: 101
8: 101
9: 101
10: 101
11: 101
aqaqbqccc
aaabbbccc
0: 97
1: 113
2: 97
3: 98
4: 98
5: 98
6: 99
7: 99
8: 99
9: 0
10: 0
11: 113
